From ecc014c46defd5d3ecb260c407b6dbf4a4374da5 Mon Sep 17 00:00:00 2001
From: Subhendu Sekhar Behera <sbehera@broadcom.com>
Date: Fri, 12 Dec 2014 15:03:33 +0530
Subject: mxlp8xx: fix multi node kernel device tree

Add I2C devices for node 1 and fix netlogic.cpumask typo.
[Based on SDK 3.2]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
Signed-off-by: Nam Ninh <nam.ninh@windriver.com>

diff --git a/arch/mips/boot/dts/xlp8xx-linux-mn-kernel.dts b/arch/mips/boot/dts/xlp8xx-linux-mn-kernel.dts
index c115972..d1d0a8c 100644
--- a/arch/mips/boot/dts/xlp8xx-linux-mn-kernel.dts
+++ b/arch/mips/boot/dts/xlp8xx-linux-mn-kernel.dts
@@ -73,16 +73,6 @@
 			clock-frequency = <32000000>;
 			interrupt-parent = <&pic0>;
 			interrupts = <30>;
-
-			rtc@68 {
-				compatible = "dallas,ds1374";
-				reg = <0x68>;
-			};
-
-			dtt@4c {
-				compatible = "national,lm90";
-				reg = <0x4c>;
-			};
 		};
 		i2c01: ocores@33000 {
 			compatible = "opencores,i2c-ocores";
@@ -282,6 +272,42 @@
 			interrupt-parent = <&pic1>;
 			interrupts = <18>;
 		};
+		i2c10: ocores@72000 {
+			compatible = "opencores,i2c-ocores";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0 0x72100 0xa00>;
+			reg-shift = <2>;
+			reg-io-width = <4>;
+			clock-frequency = <32000000>;
+			interrupt-parent = <&pic1>;
+			interrupts = <30>;
+		};
+		i2c11: ocores@73000 {
+			compatible = "opencores,i2c-ocores";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0 0x73100 0xa00>;
+			reg-shift = <2>;
+			reg-io-width = <4>;
+			clock-frequency = <32000000>;
+			interrupt-parent = <&pic1>;
+			interrupts = <31>;
+
+			rtc@68 {
+				compatible = "dallas,ds1374";
+				reg = <0x68>;
+			};
+
+			dtt@4c {
+				compatible = "national,lm90";
+				reg = <0x4c>;
+			};
+			eeprom@57 {
+				compatible = "xlpge_eeprom";
+				reg = <0x57>;
+			};
+		};
 
 		nand_flash1@79000 {
 			compatible = "netlogic,xlp-nand";
@@ -377,7 +403,7 @@
 
 	chosen {
 		bootargs = "console=ttyS0,115200 rhash_entries=1048576";
-		netlogic.cpumask = <0xffffffff>;
+		netlogic,cpumask = <0xffffffff 0xffffffff>;
 		netlogic,reserve-mem = <256>;
 		netlogic,frin-fifo-mem = <128>;
 		netlogic,spill-mem = <32>; /* max spill size : 512 MB*/
diff --git a/arch/mips/boot/dts/xlp8xx-linux-mn4-kernel.dts b/arch/mips/boot/dts/xlp8xx-linux-mn4-kernel.dts
index 6f9e6b3..59ad81f 100644
--- a/arch/mips/boot/dts/xlp8xx-linux-mn4-kernel.dts
+++ b/arch/mips/boot/dts/xlp8xx-linux-mn4-kernel.dts
@@ -73,16 +73,6 @@
 			clock-frequency = <32000000>;
 			interrupt-parent = <&pic0>;
 			interrupts = <30>;
-
-			rtc@68 {
-				compatible = "dallas,ds1374";
-				reg = <0x68>;
-			};
-
-			dtt@4c {
-				compatible = "national,lm90";
-				reg = <0x4c>;
-			};
 		};
 		i2c01: ocores@33000 {
 			compatible = "opencores,i2c-ocores";
@@ -104,6 +94,11 @@
 				compatible = "national,lm90";
 				reg = <0x4c>;
 			};
+
+			eeprom@57 {
+				compatible = "xlpge_eeprom";
+				reg = <0x57>;
+			};
 		};
 
 		nor_flash@1,0 {
@@ -278,6 +273,44 @@
 			interrupt-parent = <&pic1>;
 			interrupts = <18>;
 		};
+		i2c10: ocores@72000 {
+			compatible = "opencores,i2c-ocores";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0 0x72100 0xa00>;
+			reg-shift = <2>;
+			reg-io-width = <4>;
+			clock-frequency = <32000000>;
+			interrupt-parent = <&pic1>;
+			interrupts = <30>;
+		};
+		i2c11: ocores@73000 {
+			compatible = "opencores,i2c-ocores";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0 0x73100 0xa00>;
+			reg-shift = <2>;
+			reg-io-width = <4>;
+			clock-frequency = <32000000>;
+			interrupt-parent = <&pic1>;
+			interrupts = <31>;
+
+			rtc@68 {
+				compatible = "dallas,ds1374";
+				reg = <0x68>;
+			};
+
+			dtt@4c {
+				compatible = "national,lm90";
+				reg = <0x4c>;
+			};
+
+			eeprom@57 {
+				compatible = "xlpge_eeprom";
+				reg = <0x57>;
+			};
+		};
+
 		gpio1: xlp_gpio@74000 {
 			compatible = "netlogic,xlp-gpio";
 			reg = <0 0x74000 0x1000>;
@@ -317,7 +350,7 @@
 
 	chosen {
 		bootargs = "console=ttyS0,115200 rhash_entries=1048576";
-		netlogic.cpumask = <0xffffffff>;
+		netlogic,cpumask = <0xffffffff 0xffffffff>;
 		netlogic,reserve-mem = <256>;
 		netlogic,frin-fifo-mem = <128>;
 		netlogic,spill-mem = <32>; /* max spill size : 512 MB*/
-- 
1.7.1

